src/main/java/seedu/address/model/social/Social.java

```diff
@@ -0,0 +1,184 @@
+package seedu.address.model.social;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Represents a Social in uNivUSal.
+ */
+public class Social {
+
+    enum Socials {
+        WHATSAPP,
+        TELEGRAM,
+        EMAIL,
+        INSTAGRAM
+    }
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Socials can take any values, and it should not be blank";
+
+    /*
+     * The first character of the social link must not be a whitespace,
+     * otherwise " " (a blank string) becomes a valid input.
+     */
+    public static final String VALIDATION_REGEX = "[^\\s].*";
+
+    //links array holds the links for the socials of the person.
+    //The index of the links in the array corresponds with the enum declared.
+    //Example: links[0] should hold the link to the Whatsapp social media account.
+    private String[] links;
+
+    private Socials preferred;
+
+    /**
+     * Constructs a {@code Social}.
+     */
+    public Social() {
+        this.links = new String[4];
+    }
+
+    /**
+     * Returns true if a given string is a valid URL link.
+     */
+    public static boolean isValidSocial(String test) {
+        return test.matches(VALIDATION_REGEX);
+    }
+
+    /**
+     * Adds the Whatsapp link of the person into the links array.
+     * @param link
+     */
+    public void addWhatsapp(String link) {
+        requireNonNull(link);
+        links[0] = link;
+    }
+
+    /**
+     * Adds the Telegram link of the person into the links array.
+     * @param link
+     */
+    public void addTelegram(String link) {
+        requireNonNull(link);
+        links[1] = link;
+    }
+
+    /**
+     * Adds the Email link of the person into the links array.
+     * @param link
+     */
+    public void addEmail(String link) {
+        requireNonNull(link);
+        links[2] = link;
+    }
+
+    /**
+     * Adds the Instagram link of the person into the links array.
+     * @param link
+     */
+    public void addInstagram(String link) {
+        requireNonNull(link);
+        links[3] = link;
+    }
+
+    /**
+     * Sets the Whatsapp link to null
+     */
+    public void deleteWhatsapp() {
+        links[0] = null;
+    }
+
+    /**
+     * Sets the Telegram link to null
+     */
+    public void deleteTelegram() {
+        links[1] = null;
+    }
+
+    /**
+     * Sets the Email link to null
+     */
+    public void deleteEmail() {
+        links[2] = null;
+    }
+
+    /**
+     * Sets the Instagram link to null
+     */
+    public void deleteInstagram() {
+        links[3] = null;
+    }
+
+    /**
+     * Returns Whatsapp link of the person.
+     *
+     * @return Whatsapp link of the person.
+     */
+    public String getWhatsapp() {
```

Comment:
such as the String get methods here as they're expecting a string not null

Reply:
would be better to initialize the elements of the array to a string called "null"